edgelet/provisioning/src/provisioning.rs

```diff
@@ -134,6 +138,75 @@ impl Provision for ManualProvisioning {
     }
 }
 
+// Using PhantomData here as we need to know the associate type of Hsm in the implementation
```

Comment:
Put this comment on the field itself, not the struct.

Also a better comment would be "ExternalProvisioning is not restricted to a single HSM implementation, so it uses `PhantomData` to be generic on them."